nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/pubsub/AbstractGCPubSubProcessor.java

```diff
@@ -84,4 +112,49 @@ protected ServiceOptions getServiceOptions(ProcessContext context, GoogleCredent
         return results;
     }
 
-}
+    protected TransportChannelProvider getTransportChannelProvider(ProcessContext context) {
+        final ProxyConfiguration proxyConfiguration = ProxyConfiguration.getConfiguration(context, () -> {
+            final String proxyHost = context.getProperty(PROXY_HOST).evaluateAttributeExpressions().getValue();
+            final Integer proxyPort = context.getProperty(PROXY_PORT).evaluateAttributeExpressions().asInteger();
+            if (proxyHost != null && proxyPort != null && proxyPort > 0) {
+                final ProxyConfiguration componentProxyConfig = new ProxyConfiguration();
+                final String proxyUser = context.getProperty(HTTP_PROXY_USERNAME).evaluateAttributeExpressions().getValue();
+                final String proxyPassword = context.getProperty(HTTP_PROXY_PASSWORD).evaluateAttributeExpressions().getValue();
+                componentProxyConfig.setProxyType(Proxy.Type.HTTP);
+                componentProxyConfig.setProxyServerHost(proxyHost);
+                componentProxyConfig.setProxyServerPort(proxyPort);
+                componentProxyConfig.setProxyUserName(proxyUser);
+                componentProxyConfig.setProxyUserPassword(proxyPassword);
+                return componentProxyConfig;
+            }
+            return ProxyConfiguration.DIRECT_CONFIGURATION;
+        });
+
+        return TopicAdminSettings.defaultGrpcTransportProviderBuilder()
+                .setChannelConfigurator(new ApiFunction<ManagedChannelBuilder, ManagedChannelBuilder>() {
+                    @Override
+                    public ManagedChannelBuilder apply(ManagedChannelBuilder managedChannelBuilder) {
+                        return managedChannelBuilder.proxyDetector(
+                                new ProxyDetector() {
+                                    @Nullable
+                                    @Override
+                                    public ProxiedSocketAddress proxyFor(SocketAddress socketAddress)
+                                            throws IOException {
+                                        if (Proxy.Type.HTTP.equals(proxyConfiguration.getProxyType()) && proxyConfiguration.hasCredential()) {

```

Comment:
The `proxyConfiguration.hasCredential()` check requires the presence of a username and password, however, the `HttpConnectProxiedSocketAddress.Builder` indicates that username and password are optional and can be null.  Some HTTP proxy servers do not require authentication, so it seems like this check could be removed.